MetaMask/metamask-extension
# Disable rate limiting for signature approval requests

## Explanation

The `ApprovalController` previously rate limited all approval requests to prevent two pending requests with the same origin and type.

After adopting the `ApprovalController` to trigger the signature confirmations, this resulted in signature operations failing if there was a pending confirmation of the same signature type. In addition, the badge count would display incorrectly, and approving or rejecting duplicate confirmations would create console errors.

This has been resolved by updating the `ApprovalController` to support disabling rate limiting for specific approval types, and configuring the extension instance of the `ApprovalController` to do this for all signature approval requests. In addition, any unlikely errors arising from approval request acceptance or rejection are now caught and logged as info messages, given their minimal functional impact.

* Fixes #18553 

## Manual Testing Steps

1. Go to the test dApp.
2. Click on any of the signing buttons multiple times.
3. Verify the badge count increases and all confirmations can be approved and rejected without any console errors.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
